src/app/file-manager/file-export-modal/file-export-modal.component.ts

```diff
@@ -0,0 +1,229 @@
+import { Component, Inject, NgZone } from '@angular/core';
+import { FormBuilder, FormGroup, Form } from '@angular/forms';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';
+import { Router } from '@angular/router';
+import { FilesService } from '@app/file-manager/services/files.service';
+import { OnInit } from '@angular/core/src/metadata/lifecycle_hooks';
+import { Type } from '@app/file-manager/models/type';
+import { ItemStatus } from '@app/file-manager/models/item-status';
+import { Item } from '@app/file-manager/models/item';
+import * as Transferables from '../actions/transferables.actions';
+import { GcsService } from '@app/file-manager/services/gcs.service';
+import { Store } from '@ngrx/store';
+import { AppState } from '../dbstate/app-state';
+import { MessageService } from 'primeng/components/common/messageservice';
+import { Message } from 'primeng/components/common/api';
+import { TransferablesGridComponent } from '@app/file-manager/transferables-grid/transferables-grid.component';
+import { BucketService } from '@app/file-manager/services/bucket.service';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+
+@Component({
+  selector: 'app-file-export-modal',
+  templateUrl: './file-export-modal.component.html',
+  styleUrls: ['./file-export-modal.component.scss']
+})
+export class FileExportModalComponent implements OnInit {
```

Comment:
It looks like this duplicates a lot of the code in `FileDownloadModalComponent`. It would be better if there was a common base class that contained the shared code, to avoid this duplication.